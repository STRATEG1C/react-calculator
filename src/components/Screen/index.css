.screen-input {
    width: 100%;
    line-height: 50px;
    padding: 0 10px;
    font-size: 40px;
    text-align: right;
    border: none;
    border-bottom: 1px solid gray;
    outline: none;
}
