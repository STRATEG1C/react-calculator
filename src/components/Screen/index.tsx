import React, { ChangeEvent } from 'react';
import './index.css';

interface Props {
  text: string;
  onInput: (text: string) => void;
}

export const Screen: React.FC<Props> = (props: Props) => {
  const { text, onInput } = props;

  const onChangeHandler = (e: ChangeEvent<HTMLInputElement>) => {
    const { value } = e.target;
    onInput(value.replace(/[^0-9+\-*/]/g, ''));
  }

  return (
    <div className="screen">
      <input
        className="screen-input"
        value={text}
        onChange={onChangeHandler}
        data-testid="screen-input"
      />
    </div>
  )
}

export default Screen;

