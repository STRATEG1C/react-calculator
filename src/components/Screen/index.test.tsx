import { render } from '@testing-library/react';
import Screen from './index';
import userEvent from '@testing-library/user-event';

describe('Test screen component', () => {
  const SCREEN_TEXT = '2+2*2';
  const onInputStub = jest.fn();

  const PROPER_TEXT_TO_TEST = '1';
  const BAD_TEXT_TO_TEST = 'a';

  it('Should be rendered', () => {
    const { getByTestId } = render(<Screen text={SCREEN_TEXT} onInput={onInputStub} />);

    const screen = getByTestId('screen-input') as HTMLInputElement;

    expect(screen).toBeInTheDocument();
    expect(screen.value).toBe(SCREEN_TEXT);
    expect(screen).toMatchSnapshot();
  });

  it('Should react on input', () => {
    const { getByTestId } = render(<Screen text={SCREEN_TEXT} onInput={onInputStub} />);

    const screen = getByTestId('screen-input');

    userEvent.type(screen, PROPER_TEXT_TO_TEST);

    expect(onInputStub).toBeCalledWith(SCREEN_TEXT + PROPER_TEXT_TO_TEST);
  });

  it('Should not allow to input wrong characters', () => {
    const { getByTestId } = render(<Screen text={SCREEN_TEXT} onInput={onInputStub} />);

    const screen = getByTestId('screen-input');

    userEvent.type(screen, BAD_TEXT_TO_TEST);

    expect(onInputStub).toHaveBeenCalledWith(SCREEN_TEXT);
  });
});
