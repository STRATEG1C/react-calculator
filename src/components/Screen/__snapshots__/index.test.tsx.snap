// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test screen component Should be rendered 1`] = `
<input
  class="screen-input"
  data-testid="screen-input"
  value="2+2*2"
/>
`;
